crates/arroyo-connectors/src/kafka/mod.rs

```diff
@@ -323,7 +326,19 @@ impl Connector for KafkaConnector {
 
         let table = Self::table_from_options(options)?;
 
-        Self::from_config(self, None, name, connection, table, schema)
+        let allowed_metadata_udf_args = ["offset_id", "partition", "topic"];
+        if let Some(fields) = &metadata_fields {
+            for key in fields.keys() {
+                if !allowed_metadata_udf_args.contains(&key.as_str()) {
```

Comment:
This should be checking the value rather than the key (since the key is the field name, not the metadata name)